### ISSUE TITLE
Incorrect approximation for decimal numbers

### ISSUE THREAD

**Issue by uche-wealth** [savoirfairelinux/num2words#603]
I expected  `num2words(98_746_251_323_029.99)` to return *'ninety-eight trillion, seven hundred and forty-six billion, two hundred and fifty-one million, three hundred and twenty-three thousand and twenty-nine point nine nine'*

but it returns *'ninety-eight trillion, seven hundred and forty-six billion, two hundred and fifty-one million, three hundred and twenty-three thousand and twenty-nine point nine eight'*

I tried using the decimal module instead of float `num2words(Decimal(98_746_251_323_029.99))` but it returned the same result *'ninety-eight trillion, seven hundred and forty-six billion, two hundred and fifty-one million, three hundred and twenty-three thousand and twenty-nine point nine eight'*

Is this an approximation with the library? How do I get correct approximation for decimal values like this?

**Comment by azelcer** [comment]
The number `98_746_251_323_029.99` can't be exactly represented by IEEE 754 standard. The closest number is `98_746_251_323_029.98`. Python converts the number `98_746_251_323_029.99` to a floating point *before* calling `num2words`, thus the value the function receives is actually `98_746_251_323_029.98`. The same happens when you try to build a `Decimal` passing a float to the constructor: you are actually calling `num2words(Decimal(98_746_251_323_029.98))`. 

In this case, you should build a `Decimal`object using a string representation of the number: `num2words(Decimal("98_746_251_323_029.99"))`.

**Comment by uche-wealth** [comment]
You are right I should have passed a str arg to the Decimal object. However I still get the same result when I use `num2words(Decimal("98_746_251_323_029.99")) `

<img width="845" height="73" alt="Image" src="https://github.com/user-attachments/assets/9e0c4b27-f8dc-4074-a7dc-38e7533abd86" />

**Comment by jqueguiner** [comment]
Hi — **fixed in [num2words2 v1.0.10](https://github.com/jqueguiner/num2words2/releases/tag/v1.0.10)** ([PR #110](https://github.com/jqueguiner/num2words2/pull/110)). Pass the value as a string (or Decimal) to preserve precision at trillion scale:

```python
>>> from num2words2 import num2words
>>> num2words('98746251323029.99', lang='en')
'ninety-eight trillion, seven hundred and forty-six billion, two hundred and fifty-one million, three hundred and twenty-three thousand and twenty-nine point nine nine'
```

Float-literal input still hits the IEEE 754 limit — `98746251323029.99` becomes `≈ 98746251323029.984375` the moment Python parses it, before the library sees it. There's nothing num2words can do once that precision has been lost in the parser. Strings round-trip cleanly because they go through `Decimal` end-to-end.

Cross-language verified: same fix benefits `lang='fr' / 'es' / 'de' / ...` — anything using the base float-handling path.

---
*[num2words2](https://github.com/jqueguiner/num2words2) v1.0.10 — `pip install num2words2`. Open an issue on the fork if your case isn't covered.*
